mono/metadata/appdomain.c

```diff
@@ -2348,6 +2393,20 @@ mono_domain_assembly_search (MonoAssemblyLoadContext *alc, MonoAssembly *request
 	const MonoAssemblyNameEqFlags eq_flags = (MonoAssemblyNameEqFlags)(strong_name ? MONO_ANAME_EQ_IGNORE_CASE :
 		(MONO_ANAME_EQ_IGNORE_PUBKEY | MONO_ANAME_EQ_IGNORE_VERSION | MONO_ANAME_EQ_IGNORE_CASE));
 
+#ifdef ENABLE_NETCORE
+	mono_alc_assemblies_lock (alc);
+	for (tmp = alc->loaded_assemblies; tmp; tmp = tmp->next) {
+		ass = (MonoAssembly *)tmp->data;
+		g_assert (ass != NULL);
+		// TODO: can dynamic assemblies match here for netcore?
+		if (assembly_is_dynamic (ass) || !mono_assembly_names_equal_flags (aname, &ass->aname, eq_flags))
```

Comment:
I realize you are not changing anything here compared to the previous status quo but it made me wonder anyway. Should not this mimic the logic of `exact_sn_match` more closely? What are the rules for the name matching here in .NET Core?

Reply:
That's a fair point, and I'm not sure on the rules in Core myself, but I'm not inclined to hold this PR up for that since I have further ALC work to do that requires this PR as a base. I think for this PR, the TODO should be fine, and then I'll perhaps revisit.